checkBalanced.js

```diff
@@ -0,0 +1,96 @@
+class TreeNode {
+    constructor(data) {
+        this.data = data
+        this.left = null
+        this.right = null
+    }
+}
+
+class Tree {
+    constructor() {
+        this.head = null
+    }
+
+    add(data) {
+        const newNode = new TreeNode(data)
+        if (!this.head) {
+            this.head = newNode
+            return
+        }
+        this.addNode(this.head, newNode)
+    }
+
+    addNode(node, newNode) {
+        if (newNode.data < node.data) {
+            if (!node.left) {
+                node.left = newNode
+            } else {
+                this.addNode(node.left, newNode)
+            }
+        } else {
+            if (!node.right) {
+                node.right = newNode
+            } else {
+                this.addNode(node.right, newNode)
+            }
+        }
+    }
+
+    print() {
+        if (!this.head) {
+            console.log("Empty Tree")
+            return
+        }
+        const queue = [this.head]
+        while (queue.length) {
+            const node = queue.shift()
+            process.stdout.write(node.data + " ")
+    
+            if (node.left) {
+                queue.push(node.left)
+                process.stdout.write("Left: " + node.left.data + " ")
+            }
+    
+            if (node.right) {
+                queue.push(node.right)
+                process.stdout.write("Right: " + node.right.data + " ")
+            }
+            console.log()
+        }
+    }
+
+    
+    checkBalanced(node) {
+        if (node === null) {
```

Comment:
Because `null` is falsey in JS, you don't need to explicitly check it against `null`.
```suggestion
        if (!node) {
```